src/app/core/services/permission/permission.service.ts

```diff
@@ -1,40 +1,47 @@
 import { Injectable } from '@angular/core';
-import { permissions, actions } from 'src/app/core/constants/permissionsConstant';
+import { LocalStorageService } from '../localstorage.service';
+import { localKeys } from '../../constants/localStorage.keys';
 
 @Injectable({
   providedIn: 'root',
 })
 export class PermissionService {
-  public userPermissions: any = [
-    {
-      module: permissions.MANAGE_SESSION,
-      action: [
-        actions.ALL,
-        actions.CREATE,
-        actions.GET,
-        actions.EDIT,
-        actions.UPDATE,
-      ],
-    },
-    {
-      module: permissions.MANAGE_USER,
-      action: [actions.GET],
-    },
-  ];
 
-  fetchPermissions() {
-    // api call here
-  }
+  userPermissions:any = [];
+
+  constructor(private localStorage: LocalStorageService){}
 
-  hasPermission(permissions: any): boolean {
+  async hasPermission(permissions: any): Promise <boolean> {
+    if(!this.userPermissions.length){
+      await this.fetchPermissions().then((res) =>{
+        this.userPermissions = res;
+      })
+    }
     for (let userPermission of this.userPermissions) {
-      if (permissions && userPermission.action.length && permissions.module === userPermission.module) {
-        if (userPermission.action.includes(actions.ALL) || permissions.action.every((value) => userPermission.action.includes(value))) {
+      if (permissions && userPermission.request_type.length && permissions.module === userPermission.module) {
+        if (permissions.action.every((value: any) => userPermission.request_type.includes(value))) {
           return true;
         } else {
           return false;
         }
       }
     }
   }
+  
+  async fetchPermissions():Promise<any []> {
+    return new Promise((resolve, reject) => {
+      try {
+        this.localStorage.getLocalData(localKeys.USER_DETAILS)
+          .then(async (data) => {
+            if(data) {
+              this.userPermissions = data.permissions;
```

Comment:
you are setting this variable already in line number 17. do you need it here also?